File: integrations/aws-strands/error-code-corpus/typescript/message_helper_argument.ts
/** Sentences that reach the wire as arguments to a single-purpose builder. */
import { EventType, type BaseEvent } from "@ag-ui/core";

function resumeError(message: string): BaseEvent {
  return {
    type: EventType.RUN_ERROR,
    message,
    code: "HELPER_ARGUMENT",
  };
}

export function emit(interruptId: string): BaseEvent {
  if (!interruptId) return resumeError("The helper argument sentence.");
  return resumeError(`Interrupt ${interruptId} is unknown.`);
}

File: integrations/aws-strands/error-code-corpus/python/message_helper_argument.py
"""Sentences that reach the wire as arguments to a single-purpose builder."""

from ag_ui.core import EventType, RunErrorEvent


def resume_error(message: str) -> RunErrorEvent:
    return RunErrorEvent(
        type=EventType.RUN_ERROR,
        message=message,
        code="HELPER_ARGUMENT",
    )


def emit(interrupt_id: str) -> RunErrorEvent:
    if not interrupt_id:
        return resume_error("The helper argument sentence.")
    return resume_error(f"Interrupt {interrupt_id} is unknown.")
